homeassistant/components/mqtt/entity.py

```diff
@@ -590,6 +591,7 @@ async def cleanup_device_registry(
     entity_registry = er.async_get(hass)
     if (
         device_id
+        and device_id not in device_registry.deleted_devices
```

Comment:
Why is this check added?

Reply:
If a device based discovery is triggered to be removed and all components removed, lets say 2 components, then this check will be done twice, as we do not await anything and the actual cleanup is processed later.
From the tests it seemed this would cause issues. With single components the updates have more time to process, and this race does not occur.